src/main/java/technology/tabula/TextChunk.java

```diff
@@ -28,11 +29,100 @@ public TextChunk(List<TextElement> textElements) {
     }
 
     // used for "normalizing" RTL languages.
+    // TEMPORARY TODO REMOVE etc.
     public void reverseTextElements(){
         Collections.reverse(this.textElements);
     }
-        
-    
+
+
+    // I hate Java so bad.
+    private HashMap<Byte, String> directionalities = new HashMap<Byte, String>() {{   // BCT = bidirectional character type
```

Comment:
Both of your suggestions are now done. I didn't make the `directionalities` HashMap final -- though I did make it `static` -- because it caused a marked increase in the time it took to run the tests (over several runs, making sure not to include the compilation step). I don't know enough about Java to know whether it ought to make a difference.